src/status_im/contexts/wallet/add_address_to_watch/view.cljs

```diff
@@ -22,6 +23,10 @@
      (or (validation/eth-address? user-input)
          (validation/ens-name? user-input))) (i18n/label :t/invalid-address)))
 
+(defn- extract-address
+  [scanned-text]
+  (first (re-seq constants/regx-address-contains scanned-text)))
```

Comment:
We might want to just use `re-find` since it returns the first match.

(re-find)[https://clojuredocs.org/clojure.core/re-find]
(re-seq)[https://clojuredocs.org/clojure.core/re-seq]

Additionally, I'm wondering what happens when it's not found? is that case properly handled?

Reply:
Thanks @ulisesmac, updated the function :)

> Additionally, I'm wondering what happens when it's not found? is that case properly handled?

That should not happen as we are calling `extract-address` with `:wallet/watch-address-validated-address` sub value, and it always returns a valid address. If there's no valid address the button is disabled and `extract-address` is never called.

Reply:
Ok :+1:  thanks for the explanation